src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -447,6 +448,48 @@ internal string AdvanceThroughBalancedParentheticalExpression()
             return expressionText;
         }
 
+        /// <summary>
+        /// Get the current position in this lexer that can be used restore the lexer to this position later.
+        /// </summary>
+        /// <returns>
+        /// Returns a snapshot position used to call RestorePosition.
+        /// </returns>
+        internal ExpressionLexerPosition SnapshotPosition()
+        {
+            return new ExpressionLexerPosition(this, this.textPos, this.token);
+        }
+
+        /// <summary>
+        /// Sets the current position to the specified position.
+        /// </summary>
+        /// <param name="position">The position to restore, returned from SnapshotPosition.</param>
+        /// <remarks>
+        /// The specified position must have been retrieved by GetPostion on this instance.
+        /// </remarks>
+        internal void RestorePosition(object position)

```

Comment:
>object [](start = 38, length = 6)

can we use ExpressionLexerPosition type for position directly and skip the type casting in L471?